File: 0202-happy-number/0202-happy-number.cpp
class Solution {
public:
    bool isHappy(int n) {
        unordered_set<int> seen;

        if(n==1){
            return true;
        }

        while(n!=1 && seen.count(n)==0){
            seen.insert(n);
            string m = to_string(n);
            int sum = 0;
            for(int i =0; i<m.size(); i++){
                int a = m[i]-'0';
                sum = sum + a * a;
            }
            n = sum;
            if(n==1){
                return true;
            }
        }
        return false;
        
        
    }
};

File: 0202-happy-number/0202-happy-number.py
class Solution:
    def isHappy(self, n: int) -> bool:
        if n==1:
            return True
            
        seen = set()
        while(n!=1 and n not in seen):
            seen.add(n)
            m = str(n)
            sum = 0
            for i in range(len(m)):
                a = int(m[i])
                sum = sum + a*a

            n = sum
            if sum == 1:
                return True

        return False
        